### ISSUE TITLE
Wrong time of events

### ISSUE THREAD

**Issue by Bogmen** [cgeo/cgeo#16391]
### Describe your problem!

Die Startzeit in Events wird falsch angezeigt.
Ein Event, das um 12:00 Uhr starten soll, wird mit 00:00 Uhr angezeigt.
Ein Beispiel:
https://coord.info/GCB0PBW

### How to reproduce?

Ständig

### Actual result after these steps?

./.

### Expected result after these steps?

./.

### Reproducible

Yes

### c:geo Version

2024.11.22

### System information

```text
## System information

c:geo version: 2024.11.22

Device:
-------
- Device type: SM-N960F (crownltexx, samsung)
- Available processors: 8
- Android version: 10
- Android build: QP1A.190711.020.N960FXXU9FVH1
- Screen resolution: 1440x2792px (411x773dp)
- Pixel density: 4.5499997
- System font scale: 1.3 / used scale: 1.3
- Sailfish OS detected: false
- Google Play services: enabled - 24.46.31 (100400-699197446)
- Memory:  Available:2,6 GB, Total:7,5 GB, Threshold: 216,0 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:193
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Time zone: GMT+01:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: 10. Feb. 2023, 23:23
- Last auto backup: 25. Nov., 14:21
- Routing mode: Straight (shortest.brf / trekking.brf / car-eco.brf / (keine) / (keine))
- Live map mode: false
- Use unified map: false
- OSM multi-threading: true / threads: 4
- Map: Kombiniert (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMultiMapSource
  - Atts: belgium.map: / Map data (c) OpenStreetMap contributors /  / france.map: / Map data (c) OpenStreetMap contributors /  / germany.map: / Map data (c) OpenStreetMap contributors /  / luxembourg.map: / Map data (c) OpenStreetMap contributors /  / netherlands.map: / Map data (c) OpenStreetMap contributors
  - Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: Schwierigkeit / Gelände: * / *, Status: Gefunden=Nein ({"advanced":true,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{"special":0.0,"includeSpecial":true},"t":{"special":0.0,"includeSpecial":true}}},{"type":"status","config":{"values":["found_no"]}}]}})
- OFFLINE: Keine ({"advanced":false,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{},"t":{}}},{"type":"status","config":{}}]}})

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Anmeldung OK) / PREMIUM
   - Geocaching.com Adventure Lab
- Geocaching.com date format: dd.MM.yyyy
- Geocaching.com website language: de-DE
- Last login error on geocaching.com: communication error //  (2024-11-01 08-51)
- Last successful login on geocaching.com: 2024-11-25 19-06
- Routing: internal / BRouter installed: false

Permissions
-------
- ACCESS_COARSE_LOCATION: granted
- ACCESS_FINE_LOCATION: granted
- READ_CONTACTS: granted
- WRITE_EXTERNAL_STORAGE: DENIED
- POST_NOTIFICATIONS: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (394,9 GB free) v3 internal isDir(14 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (394,9 GB free) v3 internal isDir(14 entries)
- Geocache data: /storage/089E-ABA2/Android/data/cgeo.geocaching/files/GeocacheData (3,8 GB free) v3 external removable isDir(10112 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (394,9 GB free) v3 internal isDir(0 entries)
- Map render theme path: 
- Public Folders: #13
  - BASE: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo (av:true, files:>=4, size:>=495,2 MB, free:>=394,8 GB)
  - OFFLINE_MAPS: content://com.android.externalstorage.documents/tree/primary:cgeo/maps/document/primary:cgeo/maps (av:true, files:>=10, size:>=7,2 GB, free:>=394,8 GB)
  - OFFLINE_MAP_THEMES: content://com.android.externalstorage.documents/tree/primary:cgeo/maps/document/primary:cgeo/maps/_themes (av:true, files:>=0, size:>=0 B, free:>=394,8 GB)
  - OFFLINE_MAP_SHADING: content://com.android.externalstorage.documents/tree/primary:cgeo/maps/document/primary:cgeo/maps/_hgt (av:true, files:>=0, size:>=0 B, free:>=394,8 GB)
  - LOGFILES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/logfiles (av:true, files:>=17, size:>=1,7 MB, free:>=394,8 GB)
  - GPX: content://com.android.externalstorage.documents/tree/primary:cgeo/gpx/document/primary:cgeo/gpx (av:true, files:>=6, size:>=172,2 KB, free:>=394,8 GB)
  - BACKUP: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/backup (av:true, files:>=6, size:>=764,5 MB, free:>=394,8 GB)
  - FIELD_NOTES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/field-notes (av:true, files:>=0, size:>=0 B, free:>=394,8 GB)
  - SPOILER_IMAGES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/GeocachePhotos (av:true, files:>=2, size:>=4,0 MB, free:>=394,8 GB)
  - ROUTING_BASE: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/routing (av:true, files:>=9, size:>=155,1 MB, free:>=394,8 GB)
  - ROUTING_TILES: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/routing/segments4 (av:true, files:>=1, size:>=155,1 MB, free:>=394,8 GB)
  - WHERIGO: content://com.android.externalstorage.documents/tree/primary:cgeo/document/primary:cgeo/wherigo (av:true, files:>=0, size:>=0 B, free:>=394,8 GB)
  - TEST_FOLDER: file:///data/user/0/cgeo.geocaching/files/unittest (av:true, files:>=0, size:>=0 B, free:>=394,9 GB)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #3
  - content://com.android.externalstorage.documents/tree/primary:cgeo (29. März 2021, 10:27):RW
  - content://com.android.externalstorage.documents/tree/primary:cgeo/gpx (29. März 2021, 10:27):RW
  - content://com.android.externalstorage.documents/tree/primary:cgeo/maps (29. März 2021, 10:27):RW

Database
-------
- File: /storage/emulated/0/Android/data/cgeo.geocaching/files/databases/data (v103, Size:269,4 MB) on user storage
- Data: {cg_attributes=94334, cg_caches=13557, cg_caches_lists=17403, cg_categories=7, cg_extension=385, cg_filters=0, cg_lists=75, cg_logCount=49837, cg_logImages=27266, cg_logs=362385, cg_logs_offline=1, cg_logs_offline_images=0, cg_logs_offline_trackables=94, cg_route=0, cg_search_destination_history=0, cg_spoilers=11095, cg_trackables=2422, cg_trail_history=700, cg_variables=3547, cg_waypoints=17998}
- Extension Data: {DBEXTENSION_DOWNGRADEABLE_DBVERSION=1, DBEXTENSION_FOUNDNUM=1, DBEXTENSION_LAST_TRACKABLE_ACTION=319, DBEXTENSION_ONE_TIME_DIALOGS=6, DBEXTENSION_PENDING_DOWNLOAD=3, DBEXTENSION_POCKETQUERY_HISTORY=55}

--- End of system information ---
```

**Comment by SammysHP** [comment]
Dies sollte mit den letzten Updates bereits behoben sein. Allerdings muss der Cache noch aktualisiert werden, falls du ihn zuvor in c:geo aufgerufen hattest.

**Comment by SammysHP** [comment]
Can't reproduce with the same nightly, see screenshot above. Was the cache opened before using that version? Which timezone do you use on geocaching.com?

**Comment by rouroni1** [comment]
I've opened this geocache / event after seeing the issue reported this morning, and with the latest nightly... :( 

![TimeSettings](https://github.com/user-attachments/assets/f942c7ea-02f5-4c3d-8877-daf5eac1cc52)

**Comment by ztNFny** [comment]
> @ztNFny Any idea what's wrong here? I can't reproduce it unfortunately.

I see the problem. I tested 00:00 quite extensive but missed to test 12:00 🙁 The -12 needs to be conditional depending on PM/AM match. Will try to submit a PR later tonight.
Did i mention i hate timestrings?! This stupid change from GC alone is worth going to API....

**Comment by Bogmen** [comment]
Der Cache wurde definitiv nach dem Update gespeichert.
Eine Aktualisierung des Listings bringt nichts.

**Comment by rouroni1** [comment]
For me it's fixed with nightly 
2024.12.01-NB-fe6b291 🙂

**Comment by moving-bits** [comment]
Seems to be fixed with #16393, no further reports on support since then, thus closing here
